modules/services/osm_wikibase.js

```diff
@@ -0,0 +1,185 @@
+     * @param langCode string e.g. 'fr' for French
+     */
+    claimToValue: function(entity, property, langCode) {
+        if (!entity.claims[property]) return undefined;
+        var region = this.regionCodes[langCode];
+        var preferredPick, regionPick;
+        _forEach(entity.claims[property], function(stmt) {
+            // If exists, use value limited to the needed language (has a qualifier P26 = region)
+            // Or if not found, use the first value with the "preferred" rank
+            if (!preferredPick && stmt.rank === 'preferred') {
+                preferredPick = stmt;
+            }
+            if (stmt.qualifiers && stmt.qualifiers.P26 && stmt.qualifiers.P26[0].datavalue.value.id === region) {
+                regionPick = stmt;
+            }
+        });
+        var result = regionPick || preferredPick;
+
+        if (result) {
+            var datavalue = result.mainsnak.datavalue;
+            return datavalue.type === 'wikibase-entityid' ? datavalue.value.id : datavalue.value;
+        } else {
+            return undefined;
+        }
+    },
+
+
+    getDescription: function(entity) {
+        if (entity.descriptions) {
+            // Assume that there will be at most two languages because of
+            // how we request it: English + possibly another one.
+            // Pick non-English description if available (if we have more than one)
+            var langs = Object.keys(entity.descriptions);
+            if (langs.length) {
+                var lng = langs.length > 1 && langs[0] === 'en' ? langs[1] : langs[0];
+                return entity.descriptions[lng].value;
+            }
+        }
+        return undefined;
+    },
+
+
+    toSitelink: function(key, value) {
+        var result = value ? 'Tag:' + key + '=' + value : 'Key:' + key;
+        return result.replace(/_/g, ' ').trim();
+    },
+
+
+    getEntity: function(params, callback) {
+        var doRequest = params.debounce ? debouncedRequest : request;
+
+        var titles = [];
+        var languages = ['en'];
+        var result = {};
+        var keySitelink = this.toSitelink(params.key);
+        var tagSitelink = params.value ? this.toSitelink(params.key, params.value) : false;
+
+        if (params.langCode && params.langCode !== 'en') {
+            languages.push(params.langCode);
+        }
+
+        if (_wikibaseCache[keySitelink]) {
+            result.key = _wikibaseCache[keySitelink];
+        } else {
+            titles.push(keySitelink);
+        }
+
+        if (tagSitelink) {
+            if (_wikibaseCache[tagSitelink]) {
+                result.key = _wikibaseCache[tagSitelink];
+            } else {
+                titles.push(tagSitelink);
+            }
+        }
+
+        if (!titles.length) {
+            // Nothing to do, we already had everything in the cache
+            return callback(null, result);
+        }
+
+        var obj = {
+            action: 'wbgetentities',
```

Comment:
Do you think there’s an opportunity to merge this code with the existing Wikidata item lookup functionality somehow, given that they’re both Wikibase instances?

https://github.com/openstreetmap/iD/blob/3de03439e23b5f605c43b3667361990cd168ab94/modules/services/wikidata.js#L30-L38

Reply:
I'd rather not, at least right now.  They are 2 different wikibases.  (Also, I don't see us ever adding more).

Reply:
Not sure TBH.  They do use the same API, but they tailor to wikidata vs osmwiki structure differently. For example, osmwiki uses specific properties and qualifiers to limit the data, has a different sitelink handling, etc. Eventually it might make sense to have a separate utilities library for the common functionality, e.g. to parse claims, but I think services are sufficiently different to warrant separate code - different properties, different lookup methods.

Reply:
Actually we could probably get rid of `services/wikipedia.js` and fold what that does into `services/wikidata.js` (but again lets do that another time)